### PR TITLE
Allow overriding asset presets

### LINKED ISSUE #19485: Allow overriding params when using `&key=x` in assets endpoint
### Describe the Improvement

If your project is configured to allow any preset, you should be able to override individual options when using `key` in `/assets`, like `key=thumbnail&width=500` should use the configured preset for everything, but then use `500` for width.

### PR DESCRIPTION
**Pull request by br41nslug**
Fixes #19485 

## Scope

What's changed:

- Allows overwriting asset transformation properties combined with a preset
```
http://localhost:8055/assets/ab457687-ff06-40de-b400-bce788934163
  ?key=test
  &width=500
  &quality=50
  &format=jpg
```
This will only be accepted when all asset transformations are allowed in the project settings else this will error. 

## Potential Risks / Drawbacks

- This feature can possibly be used to cache bust

## Review Notes / Questions

- I would like to lorem ipsum

### CODE REVIEW

File: api/src/controllers/assets.ts [hunk 1]
```diff
@@ -152,11 +156,13 @@ router.get(
...
-		const transformationParams: TransformationParams = res.locals['transformation'].key
+		const transformationPresets = res.locals['transformation'].key
 			? (res.locals['shortcuts'] as TransformationParams[]).find(
 					(transformation) => transformation['key'] === res.locals['transformation'].key
-			  )
-			: res.locals['transformation'];
+			  ) ?? {}
+			: {};
+
+		const transformationParams: TransformationParams = { ...transformationPresets, ...res.locals['transformation'] };
```
Comment (DanielBiegler) [inline api/src/controllers/assets.ts:165]: We could make that more compact with just

```ts
		const transformationParams: TransformationParams = {
			...(res.locals['shortcuts'] as TransformationParams[]).find(
				(transformation) => transformation['key'] === res.locals['transformation'].key
			),
			...res.locals['transformation'],
		};
```

Comment (br41nslug) [reply api/src/controllers/assets.ts:165]: `shurtcuts.find` can return undefined so trying to expand that with `...` directly may error.

Comment (DanielBiegler) [reply api/src/controllers/assets.ts:165]: For Arrays yes but Objects do work with undefined

```js
{ ...undefined, ...{thisworks:123} } // {thisworks: 123}
```

Doesnt throw wheras

```js
[ ...undefined, ...[1,2] ] // Uncaught TypeError: undefined is not iterable
```

does throw

Comment (br41nslug) [reply api/src/controllers/assets.ts:165]: Ah yeah right, `null` and `undefined` only fail on iterating not for object spreading 😂 what a language, they're both considered objects arent they.

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**